src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -548,7 +548,7 @@ JNI_METHOD(void, setAttestationTrustStoreDelegate)
 
 JNI_METHOD(void, startOTAProvider)(JNIEnv * env, jobject self, jlong handle, jobject otaProviderDelegate)
 {
-#if CHIP_DEVICE_CONFIG_DYNAMIC_SERVER
+#ifdef CHIP_DEVICE_CONFIG_DYNAMIC_SERVER
```

Comment:
Maybe better define it as `CHIP_DEVICE_CONFIG_DYNAMIC_SERVER=1` (with a fallback to `=0` in the CHIPConfig.h)? It will be consistent with other defines and the `-Wundef` will catch errors.